### ISSUE TITLE
[expo-file-system] How to rename a file?

### ISSUE THREAD

**Issue by gattigaga** [expo/expo#25344]
### Minimal reproducible example

https://snack.expo.dev/@gattigaga/expo-file-system-rename-a-file

## Platform

OS Android 12

## Expected Behavior

By using either `FileSystem.copyAsync()` or `FileSystem.moveAsync()` method, should be able to copy or move the file to the new location with new filename.

```javascript
await FileSystem.copyAsync({
  from: "file:///data/user/0/host.exp.exponent/files/statuses/old_file.jpeg",
  to: `file:///data/user/0/host.exp.exponent/files/statuses/new_file.jpeg`,
});
```

## Actually Happening

By using either `FileSystem.copyAsync()` or `FileSystem.moveAsync()` method, it threat new filename as the new directory instead of file.

```javascript
await FileSystem.copyAsync({
  from: "file:///data/user/0/host.exp.exponent/files/statuses/old_file.jpeg",
  to: `file:///data/user/0/host.exp.exponent/files/statuses/new_file.jpeg`,
});
```

Then the result is:
file:///data/user/0/host.exp.exponent/files/statuses/new_file.jpeg/old_file.jpeg

How come the new_file.jpeg is not a file but become directory?

I was checking using `FileSystem.getInfoAsync()` method and it shows `isDirectory: true` for this path:
file:///data/user/0/host.exp.exponent/files/statuses/new_file.jpeg

## Summary

Basically I want to rename a file but seems in the expo-file-system documentation there's no method used for renaming a file. So is there any solutions for this?

### Environment

expo-env-info 1.0.5 environment info:
    System:
      Shell: 5.8.1 - /usr/bin/zsh
    Binaries:
      Node: 16.20.2 - /run/user/1000/fnm_multishells/8369_1699760535082/bin/node
      npm: 8.19.4 - /run/user/1000/fnm_multishells/8369_1699760535082/bin/npm
      Watchman: 2023.10.09.00 - /home/linuxbrew/.linuxbrew/bin/watchman
    IDEs:
      Android Studio: AI-223.8836.35.2231.10811636
    npmPackages:
      expo: ~49.0.15 => 49.0.16 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.6 => 0.72.6 
    npmGlobalPackages:
      eas-cli: 5.6.0
    Expo Workflow: managed

**Comment by Kudo** [comment]
@alanjhughes could you help to take a look for this issue?

i guess we had inconsistent input in `transformFilesFromSAF()`
`transformFilesFromSAF()` expects its parameter as `outputDir`: https://github.com/expo/expo/blob/3d96d546792f56ea70fb497de5020d148995a7c5/packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt#L822
but we passed the argument as just file: https://github.com/expo/expo/blob/3d96d546792f56ea70fb497de5020d148995a7c5/packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt#L278-L279

**Comment by gattigaga** [comment]
Thank you guys! :)
